docs/DeveloperGuide.md

```diff
@@ -1,38 +1,308 @@
+
+![Favourite Command Class Diagram](./images/FavouriteCommand_Class.png)
+
+The following sequence diagram illustrates the flow of the program to read in the user input, parse the user input, check if it is a valid favourite command, and execute the relevant favourite command.
+
+![Favourite Command Sequence Diagram](./images/FavouriteCommand_Sequence.png)
+
+### 1.7 UserModuleMappingList
+
+#### 1.7.1 UserModuleMapping
+
+The UserModuleMapping class aims to bridge 2 modules (one from NUS and one from a Partner University). It is initialized by a constructor
+requiring the module code, title, and university name for both NUS and partner university. This aims to simulate a real life example of users
+mapping an NUS module to a foreign university's.
+
+#### 1.7.2 UserModuleMappingList
+
+The UserModuleMappingList class consists of stores a list of UserModuleMapping in an ArrayList. Users are able to add new modules using the `addModule`
+function, search for matching modules using `findModuleByCode` and delete modules using `deleteModule`. Additionally, users are also able to search
+for modules using keywords to filter out the modules currently stored that are related using `findModuleByTitle`
+
+The following class diagram illustrates the relationship between UserModuleMappingList and UserModuleMapping.
+
+![UserModuleMappingList Diagram](./images/UserModuleMapping_Class.png)
+
+### 1.8 UserUniversityListManager
+
+#### 1.8.1 UserUniversityList
+
+The UserUniversityList class stores 2 important things the `universityName` and a list of UserModuleMapping under `myModules`
+Each list is identified using the universityName. Users can only create 1 list for each partner university and this is managed by the 
+UserUniversityListManager in 1.8.2. A notable function in UserUniversityList is `setFavourite` which will be used in the favourite function
+to help users manage multiple lists and note down their favourites
+
+#### 1.8.2 UserUniversityListManager
+
+The UserUniversityListManager manages a hashmap of lists, with the `universityName` as the key. This prevents duplicates and unnecessary space
+wastage. HashMap is also an efficient data structure to obtain the UserUniversityList as the value in constant time. The UserUniversityListManager
+has the notable functions `addModule` and `deleteModule` which allows users to add and delete module in a specific list. It also has the `addFavourite`
+and `deleteFavourite` function which helps the users to organise their lists. 
+
+The following class diagram illustrates the relationship between UserUniversityListManager and UserUniversityList as well as other relevant classes.
+
+![UserUniversityListManager Diagram](./images/UserUniversityListManager_Class.png)
+
+The following sequence diagram helps explain the key steps behind the main functions in UserUniversityListManager.
+
+![UserUniversityListManager Sequence](./images/UserUniversityListManager_Sequence.png)
```

Comment:
Would it be better if you can split each main function into separate sequence diagrams? I think it would be more helpful if the diagrams are split apart with some more explanation added for each of them.